src/Sylius/Component/Core/spec/Promotion/Applicator/OrderItemPromotionAdjustmentsApplicatorSpec.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Applicator;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Promotion\Applicator\OrderItemPromotionAdjustmentsApplicator;
+use Sylius\Component\Core\Promotion\Applicator\OrderItemPromotionAdjustmentsApplicatorInterface;
+use Sylius\Component\Order\Factory\AdjustmentFactoryInterface;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ * @author Gorka Laucirica <gorka.lauzirika@gmail.com>
+ */
+final class OrderItemPromotionAdjustmentsApplicatorSpec extends ObjectBehavior
+{
+    function let(
+        AdjustmentFactoryInterface $adjustmentFactory
+    ) {
+        $this->beConstructedWith($adjustmentFactory);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(OrderItemPromotionAdjustmentsApplicator::class);
+    }
+
+    function it_implements_an_units_promotion_adjustments_applicator_interface()
+    {
+        $this->shouldImplement(OrderItemPromotionAdjustmentsApplicatorInterface::class);
+    }
+
+    function it_applies_promotion_adjustments_on_all_units_of_given_order_item(
+        OrderItemInterface $orderItem,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion,
+        AdjustmentFactoryInterface $adjustmentFactory,
+        AdjustmentInterface $firstAdjustment,
+        AdjustmentInterface $secondAdjustment
+    ) {
+        $promoName = 'Winter guns promotion!';
+        $promoCode = 'WINTER_GUNS_PROMOTION';
```

Comment:
I think it's not needed to extract this strings to variables. I know they're used in many places in this function and normally they should be extracted, but specification should be read from top to bottom, without thinking about some previously defined parameters.